src/app/components/CharacterTable.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState, useRef, useEffect } from 'react';
+import type { ColumnType, TableProps } from 'antd/lib/table';
+import { SearchOutlined } from '@ant-design/icons';
+import type { FilterValue } from 'antd/lib/table/interface';
+import type { InputRef } from 'antd';
+import {
+  Table,
+  Input,
+  Button,
+  Space,
+  Spin,
+} from 'antd';
+
+import { ITable, ICharactersTable } from 'utils/types';
+import columns from 'utils/columns/characterColumns';
+import CharacterModal from './Modal/CharacterModal';
+
+type Characters = keyof ICharactersTable;
+
+const DashBoardTable: React.FC<ITable> = (props: ITable) => {
+  const {
+    moreLoading,
+    data,
+    pagination,
+    setFilter,
+    setPagination,
+  } = props;
+  const searchInput = useRef<InputRef>(null);
+  const [id, setID] = useState<string>('');
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [modifiedColumns, setModifiedColumns] = useState<ICharactersTable[]>();
+
+  const onChange: TableProps<ICharactersTable>['onChange'] = (newPagination, filters: Record<string, FilterValue | null>) => {
+    const filterObj: {[key: string]: string} = {};
+    Object.keys(filters).forEach((key) => {
+      const filterValue = filters[key];
+      if (filterValue !== null) {
+        if (Array.isArray(filterValue)) {
+          filterObj[key] = filterValue[0].toString();
+        }
+      }
+    });
+    setFilter(filterObj);
+    setPagination(newPagination);
+  };
+
+  const showModal = (
+    e: React.MouseEvent<HTMLElement>,
+    index: string,
+  ) => {
+    e.stopPropagation();
+    setID(index);
+    setIsModalVisible(true);
+  };
+
+  const getColumnSearchProps = (dataIndex: Characters): ColumnType<ICharactersTable> => ({
+    filterDropdown: ({
+      setSelectedKeys,
+      selectedKeys,
+      confirm,
+      clearFilters,
+    }) => (
+      <div style={{ padding: 8 }}>
+        <Input
+          ref={searchInput}
+          placeholder={`Search ${dataIndex}`}
+          value={selectedKeys[0]}
+          onChange={(e) => setSelectedKeys(e.target.value ? [e.target.value] : [])}
+          onPressEnter={() => confirm()}
```

Comment:
is there a reason to have this arrow function wrapping the confirm call? are there any pros / cons of this approach?

Reply:
When we use only confirm function, it will use the void as a param.
But, As "onPressEnter" function is the event function, it will not get the void type as a param.
So, I've used the arrow function.
Arrow function's main benefit is the shorter syntax and thus requires less code.
I think that It will increase the code quality

Reply:
https://reactjs.org/docs/faq-functions.html#arrow-function-in-render in case you also want to check what the cons of this approach are

Reply:
Okay. In this case, it could also affect performance optimization.
Thanks for your advice
I will reference in the future